src/js/builtins.d.ts

```diff
@@ -407,6 +407,8 @@ declare function $streamErrored(): TODO;
 declare function $streamReadable(): TODO;
 declare function $streamWaiting(): TODO;
 declare function $streamWritable(): TODO;
+declare function $stringIndexOfInternal(searchString: string, position?: number): TODO;
+declare function $stringSubstring(indexStart: number, indexEnd?: number): TODO;
```

Comment:
nice finds. do these return number and string respectively. could add that.

this file was mostly a stub based on what i thought existed. i should clean it up to be exaclty what is available and fill out the types

Reply:
> nice finds. do these return number and string respectively. could add that.

yes :) they return identical to the builtins. I will update their definitions.